### PR TITLE
Issue #16 Registration per role

### PR DESCRIPTION
**Pull request by rollki**

### CODE REVIEW

File: modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationStatus.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      $container->get('role.control_manager'),
+      $container->get('role_registration.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function attachElement(&$form, RoleInterface $role) {
+    $registration_third_party_settings = $this->roleRegistrationManager->getRegistrationThirdPartySettings($role);
```
Comment (mariiadenpavli) [inline modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationStatus.php:63]: '#default_value' => $role->getThirdPartySetting(RoleControlManagerInterface::MODULE_NAME, $plugin_id),

File: modules/role_registration/role_registration.info.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name: Role registration
+type: module
+description: Create registraion pages based on roles and user display modes.
+package: User
+core: '8.x'
```
Comment (ozin7) [inline modules/role_registration/role_registration.info.yml:5]: Add dependency on `role`

File: modules/role_registration/src/Controller/UserPagesController.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    );
+  }
+
+  /**
+   * @param \Drupal\role_registration\Controller\string $role_id
+   *
+   * @return array
+   */
+  public function registerPage($role_id) {
+    $role = Role::load($role_id);
```
Comment (ozin7) [inline modules/role_registration/src/Controller/UserPagesController.php]: Use entity type manager to load the role entity.

File: modules/role_registration/src/Controller/UserPagesController.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+    $form_display = $third_party_settings['account_registration_form_mode'];
+    // Be sure definition is up to date.
+    /** @var \Drupal\Core\Entity\ContentEntityType $userEntityDefinition */
+    $userEntityDefinition = $this->entityTypeManager->getDefinition('user');
+    if (!array_key_exists($form_display, $userEntityDefinition->getHandlerClasses()['form'])) {
+      $this->entityTypeManager->clearCachedDefinitions();
+    }
+
+    $entity = User::create();
```
Comment (ozin7) [inline modules/role_registration/src/Controller/UserPagesController.php]: Use entity type manager to create a user.

File: modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationFormMode.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function attachElement(&$form, RoleInterface $role) {
+    $registration_third_party_settings = $this->roleRegistrationManager->getRegistrationThirdPartySettings($role);
+    $plugin_id = $this->getPluginId();
+
+    $form['account']['registration'][$plugin_id] = [
+      '#type' => 'select',
+      '#title' => t('Registration Form mode'),
```
Comment (ozin7) [inline modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationFormMode.php]: $this->t()

File: modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationFormMode.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      '#options' => $this->roleControlManager->getUserFormModesOptions(),
+      '#description' => t('Select which form mode to use on the user registration page'),
```
Comment (ozin7) [inline modules/role_registration/src/Plugin/Role/RoleConfigElement/RegistrationFormMode.php]: $this->t()

File: src/Service/RoleControlManager.php [hunk 7]
```diff
@@ -128,4 +142,16 @@ public function getRoleThirdPartySetting(AccountInterface $user, string $config)
     return $settings[$config] ?? NULL;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getUserFormModesOptions() {
+    $user_form_modes = $this->entityDisplayRepository->getFormModeOptionsByBundle('user', 'user');
+    $user_form_modes_options = ['default' => 'Default'];
```
Comment (ozin7) [inline src/Service/RoleControlManager.php]: add $this-t("Default");

**Review by ozin7 — CHANGES_REQUESTED** [review]

**Pull request merged**